exome_alignment.cwl

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: Workflow
+label: "exome alignment with qc"
+requirements:
+    - class: SubworkflowFeatureRequirement
+inputs:
+    reference:
+        type: File
+        secondaryFiles: [.fai, .bwt, .sa, .ann, .amb, .pac, ^.dict, .alt]
+    bams:
+        type: File[]
+    readgroups:
+        type: string[]
+    mills:
+        type: File
+        secondaryFiles: [.tbi]
+    known_indels:
+        type: File
+        secondaryFiles: [.tbi]
+    dbsnp:
+        type: File
+        secondaryFiles: [.tbi]
+    bait_intervals:
+        type: File
+    target_intervals:
+        type: File
+    omni_vcf:
+        type: File
+        secondaryFiles: [.tbi]
+outputs:
+    bam:
+        type: File
+        outputSource: alignment/bqsr_bam
+    insert_size_metrics:
+        type: File
+        outputSource: qc/insert_size_metrics
+    alignment_summary_metrics:
+        type: File
+        outputSource: qc/alignment_summary_metrics
+    hs_metrics:
+        type: File
+        outputSource: qc/hs_metrics
+    per_target_coverage_metrics:
+        type: File
+        outputSource: qc/per_target_coverage_metrics
+    flagstats:
+        type: File
+        outputSource: qc/flagstats
+    verify_bam_id_metrics:
+        type: File
+        outputSource: qc/verify_bam_id_metrics
+    verify_bam_id_depth:
+        type: File
+        outputSource: qc/verify_bam_id_depth
+steps:
+    alignment:
+        run: unaligned_bam_to_bqsr/workflow.cwl
+        in:
+            reference: reference
+            bams: bams
+            readgroups: readgroups
+            mills: mills
+            known_indels: known_indels
+            dbsnp: dbsnp
+        out: [bqsr_bam]
```

Comment:
The output of `unaligned_bam_to_bqsr/workflow.cwl` is named `final_bam`.

(It's a bummer that `cwltool --validate` doesn't check this--for subworkflows it only looks that the named `.cwl` file exists!)